src/Sulu/Component/Content/Types/BlockContentType.php

```diff
@@ -240,18 +240,20 @@ private function doWrite(
                     $isImport
                 );
 
-                $this->writeProperty(
-                    $settingsProperty,
-                    $property,
-                    \json_encode($blockPropertyType->getSettings()),
-                    $i,
-                    $node,
-                    $userId,
-                    $webspaceKey,
-                    $languageCode,
-                    $segmentKey,
-                    $isImport
-                );
+                if (\is_array($blockSettings = $blockPropertyType->getSettings())) {
+                    $this->writeProperty(
```

Comment:
We should not assign variables inside the if statement:

```suggestion
                $blockSettings = $blockPropertyType->getSettings();
                if (\is_array($blockSettings)) {
                    $this->writeProperty(
```